### PR TITLE
Replace IOUtil.readFully() with InputStream.readAllBytes  [HZ-3145]

### PR DESCRIPTION
**Pull request by orcunc**
Java 11 already provides the same functionality

Jira : https://hazelcast.atlassian.net/browse/HZ-3145

Breaking changes (list specific methods/types/messages):
* API
* client protocol format
* serialized form
* snapshot format

Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Label `Add to Release Notes` or `Not Release Notes content` set
- [x] Request reviewers if possible

### CODE REVIEW

**Review by JackPGreen — APPROVED** [review]

File: hazelcast/src/test/java/com/hazelcast/client/config/XmlYamlClientConfigBuilderEqualsTest.java [hunk 1]
```diff
@@ -116,8 +115,10 @@ public void testFullClientFailoverConfig() throws Exception {
     }
 
     private String readResourceToString(String resource) throws IOException {
-        InputStream xmlInputStream = getClass().getClassLoader().getResourceAsStream(resource);
-        return new String(IOUtil.toByteArray(xmlInputStream));
+        try (InputStream xmlInputStream = getClass().getClassLoader().getResourceAsStream(resource)) {
```
Comment (JackPGreen) [inline hazelcast/src/test/java/com/hazelcast/client/config/XmlYamlClientConfigBuilderEqualsTest.java]: You could replace this with [this](https://stackoverflow.com/a/6068228) and avoid a `byte[]` entirely.

Comment (orcunc) [reply hazelcast/src/test/java/com/hazelcast/client/config/XmlYamlClientConfigBuilderEqualsTest.java]: I think you are suggesting using Guava. I don't think we need an extra dependency for this. We are trying to use a little extra dependency as possible. Thanks for the suggestion anyway 👍

Comment (JackPGreen) [reply hazelcast/src/test/java/com/hazelcast/client/config/XmlYamlClientConfigBuilderEqualsTest.java]: > I think you are suggesting using Guava. I don't think we need an extra dependency for this. We are trying to use a little extra dependency as possible. Thanks for the suggestion anyway 👍

Yes, it’s Guava. I think its already a dependency on the project which is why I suggested it.

**Comment by orcunc** [comment]
run-lab-run

File: hazelcast/src/test/java/com/hazelcast/client/config/XmlYamlClientConfigBuilderEqualsTest.java (hunk 1 above)
Comment (orcunc) [reply hazelcast/src/test/java/com/hazelcast/client/config/XmlYamlClientConfigBuilderEqualsTest.java]: Guava is not our direct dependency. It is declared under dependencyManagent. In the build system there is a maven-enforcer-plugin. It enforces all transitive dependencies but have the same version. Sometimes there is conflict and we add an entry to dependencyManagent to fix the version used.

**Pull request merged**